src/components/shared/chart/Viewport.js

```diff
@@ -63,28 +88,15 @@ require('./Viewport.css');
  * viewportRight += mouseMoveDelta * unitPixel
  * viewportLeft += mouseMoveDelta * unitPixel
  **/
-export default function withChartViewport<T>(WrappedComponent: ReactClass<T>) {
-  class ChartViewport extends PureComponent {
-    props: Props;
+export default function withChartViewport<Props: ViewportProps>(
+  WrappedComponent: React.ComponentType<InjectedProps & $Supertype<Props>>
+): React.ComponentType<Props> {
```

Comment:
Here is the rationale:
* `Props` are the props for the returned component (it means: they are the props that the _user_ for this component will need to specify). From the generic definition `<Props: ViewportProps>`, they must include the properties consumed par this HOC.
* The argument, which is the augmented component (the `WrappedComponent`), needs to accept both the `InjectedProps` and some supertype of `Props` -- which means an object will some properties of `Props` but not all. What it means to Flow which goes (sometimes) backwards is: from the props of `WrappedComponent`, take out `InjectedProps`, and make it part of `Props`; so `Props` will need to hold both `ViewportProps` as said earlier _and_ the props from `WrappedComponent` that aren't `Injectedprops`. Which is exactly what we want :)

Maybe this should be a comment, what do you think ?